### PR TITLE
ENH: add `lowercase` keyword to `Sequence`.

### PR DESCRIPTION
**Pull request by ebolyen**
Behaves just like `DNA`, `RNA`, and `Protein`

### CODE REVIEW

**Comment by RNAer** [comment]
cool. I will update genbank.py with that for now or after it is merged?

**Comment by ebolyen** [comment]
@RNAer after merge, should be by the end of today I expect.

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +28,7 @@
 * Doctests are now written in Python 3.
 * ``make test`` now validates MANIFEST.in using [check-manifest](https://github.com/mgedmin/check-manifest). ([#461](https://github.com/biocore/scikit-bio/issues/461))
 * Many new alpha diversity equations added to ``skbio.diversity.alpha`` documentation. ([#321](https://github.com/biocore/scikit-bio/issues/321))
+* Order of `lowercase` and `validate` keywords swapped in `skbio.sequence.IUPACSequence`.
```
Comment (jairideout) [inline CHANGELOG.md]: Can you note the concrete user-facing classes this affects?

File: skbio/sequence/_sequence.py [hunk 2]
```diff
@@ -520,8 +524,8 @@ def _string(self):
         return self._bytes.tostring()
 
     @stable(as_of="0.4.0")
-    def __init__(self, sequence, metadata=None,
-                 positional_metadata=None):
+    def __init__(self, sequence, metadata=None, positional_metadata=None,
+                 lowercase=False):
```
Comment (jairideout) [inline skbio/sequence/_sequence.py]: update the class docstring, `Parameters` section

Comment (jairideout) [inline skbio/sequence/_sequence.py]: the order of constructor parameters changed, please update DNA, RNA, and Protein class docstrings' Parameters section

**Comment by jairideout** [comment]
This is ready for merge once tests finish. @ebolyen feel free to merge yourself if you need this urgently. Otherwise I will merge later tonight when I'm back online.

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@
 * Added ``skbio.util.RepresentationWarning`` for warning about substitutions, assumptions, or particular alterations that were made for the successful completion of a process.
 * ``TreeNode.tip_tip_distances`` now supports nodes without an associated length. In this case, a length of 0.0 is assumed and an ``skbio.util.RepresentationWarning`` is raised. Previous behavior was to raise a ``NoLengthError``. ([#791](https://github.com/biocore/scikit-bio/issues/791))
 * ``DistanceMatrix`` now has a new constructor method called `from_iterable`.
+* ``Sequence`` now accepts ``lowercase`` keyword like ``DNA`` and others. Updated ``fasta``, ``fastq``, and ``qseq`` readers/writers for ``Sequence`` to reflect this.
```
Comment (jairideout) [inline CHANGELOG.md]: also need to mention the addition of `Sequence.lowercase` method

**Comment by jairideout** [comment]
Noticed one more changelog mention I forgot about yesterday, sorry. Once that's added this is ready for merge (tests don't need to run).

**Pull request merged**